### DISCUSSION TITLE
OKane 2022 with another SEI model

### DISCUSSION THREAD

**Discussion by iurkens** [pybamm-team/PyBaMM#2446] [Q&A] [answered]
Is it possible to use another SEI model than the solvent-diffusion limited one when using Okane's 2022 parameter set? 
I'd like to use the interstitial-diffusion limited one. When I try to do this, the solver is unable to solve the equations, and I don't know why.

With kind regards
![image](https://user-images.githubusercontent.com/116073555/200903553-e0becb01-d399-474d-b8ea-e344122adfb9.png)

**Comment by RuiheLi** [comment, marked as answer]
Hi, 

This is because the interstitial-diffusion limited SEI model must have inner SEI layer, whereas the Okane's 2022 parameter set sets them to zero. You just need to change the following parameters to the same value of the example SEI parameter set:

1. Inner SEI reaction proportion: 0.5;
2. Initial inner SEI thickness [m]: 2.5e-9; 
3. Outer SEI reaction proportion: 0.5;
4. Initial outer SEI thickness [m]: 2.5e-9; 

For more information, you can check here: https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/models/submodels/interface/sei/sei_growth.py#:~:text=elif%20self.options%5B%22SEI%22%5D%20%3D%3D%20%22interstitial,prefactor%20*%20delta_phi)%20/%20(C_sei%20*%20L_sei_inner)

**Comment by iurkens** [comment]
Hello

It works, thank you for your the help! I have an additional question:

 Do you think it is possible to use the particle model: "swelling and cracking" in combination with the SEI interstitial-diffusion model? It doesn't work now, even with the SEI on cracks disabled. 

With kind regards

**Reply by RuiheLi** [reply]
It should be compatiable. But crack model is something out of my knowledge. Maybe you can post your script or give more information about the errors?

**Reply by iurkens** [reply]
Hi

This my script (nothing special) and the outputs.

With kind regards

![](https://user-images.githubusercontent.com/116073555/201155719-dfb6f0df-b031-443e-9ffe-7bc1a4d5686a.png)
![](https://user-images.githubusercontent.com/116073555/201155546-d56eead0-f669-470d-9082-3c4ff3242e67.png)

**Reply by RuiheLi** [reply]
The default voltage cut off for OKANE2022 parameter is 4.2V and 2.5V, as seen [here](https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/input/parameters/lithium_ion/OKane2022.py#:~:text=%22Lower%20voltage%20cut,%3A%204.2%2C): 

So maybe you can reduce the voltage range in the experiment as a start, like charge until 4.0V and discharge until 2.7V
